### PR TITLE
Finalize split preparation-1

### PR DESCRIPTION
**Pull request by ronald-cron-arm**
## Description
Progresses #9624 

## PR checklist
- [x] **changelog** not required because: no change from the user perspective
- [x] **development PR** here
- [x] **framework PR** not required
- [x] **3.6 PR** not required because: 4.0 only work 
- [x] **2.28 PR** not required because: 4.0 only work 
- **tests**  not required because: just changing the way things are done

### CODE REVIEW

**Comment by ronald-cron-arm** [comment]
ABI-API-checking job falling due to the move of timing.c to the mbedtls library.

**Comment by ronald-cron-arm** [comment]
Straightforward rebase on top of latest development to resolve the conflicts.

File: tf-psa-crypto/scripts/generate_driver_wrappers.py [hunk 1]
```diff
@@ -179,6 +179,8 @@ def main() -> int:
     args = parser.parse_args()
 
     project_root = os.path.abspath(args.project_root)
+    if build_tree.looks_like_mbedtls_root(project_root):
```
Comment (minosgalanakis) [inline tf-psa-crypto/scripts/generate_driver_wrappers.py:182]: I take it that is to allow calling the scripts from the library root as well as the new location?

File: tf-psa-crypto/scripts/framework_scripts_path.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+"""Add our Python library directory to the module search path.
```
Comment (minosgalanakis) [inline tf-psa-crypto/scripts/framework_scripts_path.py:1]: Any reason we are not doing this with an __init__.py file?

**Review by minosgalanakis — APPROVED** [review]
looks good, two non blocking questions only.

File: tf-psa-crypto/scripts/generate_driver_wrappers.py (hunk 1 above)
Comment (ronald-cron-arm) [reply tf-psa-crypto/scripts/generate_driver_wrappers.py:182]: We need to keep supporting the call from mbedtls for the make build system.

**Review by gabor-mezei-arm — APPROVED** [review]
Looks good to me

**Pull request merged**